File: app/src/main/java/be/vergauwen/simon/cleanmvp/MVPApplication.java
package be.vergauwen.simon.cleanmvp;

import android.app.Application;
import be.vergauwen.simon.cleanmvp.di.ApplicationComponent;
import be.vergauwen.simon.cleanmvp.di.ApplicationModule;
import be.vergauwen.simon.cleanmvp.di.DaggerApplicationComponent;

public class MVPApplication extends Application {

    private static ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerApplicationComponent.builder()
            .applicationModule(new ApplicationModule(this))
            .build();
    }

    public static ApplicationComponent getComponent() {
        return component;
    }
}

File: app/src/main/kotlin/be/vergauwen/simon/MVPApplication.kt
package be.vergauwen.simon

import android.app.Application
import be.vergauwen.simon.di.ApplicationComponent
import be.vergauwen.simon.di.ApplicationModule
import be.vergauwen.simon.di.DaggerApplicationComponent

internal class MVPApplication : Application() {

  companion object {
    lateinit var component: ApplicationComponent
  }

  override fun onCreate() {
    super.onCreate()

    component = DaggerApplicationComponent.builder()
        .applicationModule(ApplicationModule(this))
        .build()
  }
}
